### PR TITLE
feat(about): add dynamic numbers in about page

### PR DESCRIPTION
**Pull request by Slashgear**
## 🤔 Why do you want to make those changes?

Add numbers to about page dynamically fetched from meetup api

## 🧑‍🔬 How did you make them?

- add new component and fetch data dynamically

## 🧪 How to check them?

- check about page

### CODE REVIEW

File: modules/meetup/queries/numbers.api.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+const getAge = () => {
+  var today = new Date();
+  var birthDate = new Date('2011-10-25');
+  var age = today.getFullYear() - birthDate.getFullYear();
+  var m = today.getMonth() - birthDate.getMonth();
+  if (m < 0 || (m === 0 && today.getDate() < birthDate.getDate())) {
+    age--;
+  }
+  return age;
+};
```
Comment (mathieumure) [inline modules/meetup/queries/numbers.api.ts:39]: I would add explicite return type to improve readibility and use moderne constant definition 😉 
```suggestion
const getAge = (): number => {
  const today = new Date();
  const birthDate = new Date('2011-10-25');
  let age = today.getFullYear() - birthDate.getFullYear();
  const m = today.getMonth() - birthDate.getMonth();
  if (m < 0 || (m === 0 && today.getDate() < birthDate.getDate())) {
    age--;
  }
  return age;
};
```

File: app/a-propos/page.tsx [hunk 2]
```diff
@@ -5,7 +5,7 @@ import Content from './content.mdx';
 export default function APropos() {
   return <Content />;
 }
-
+export const revalidate = 3600;
```
Comment (mathieumure) [inline app/a-propos/page.tsx:8]: ℹ️ For other reviewer this constant is a convention in [Next](https://nextjs.org/docs/app/building-your-application/data-fetching/fetching-caching-and-revalidating#time-based-revalidation)

File: modules/meetup/queries/numbers.api.ts [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+      value: numberOfEvents,
+      text: 'Événements organisés',
+    },
+    {
+      value: 3400,
+      text: 'Inscrits sur Meetup',
+    },
+    {
+      value: getAge(),
+      text: 'Notre age',
```
Comment (mathieumure) [inline modules/meetup/queries/numbers.api.ts:69]: suggestion: What about this wording instead ?
```suggestion
      text: 'Années d'activité',
```

File: modules/meetup/queries/numbers.api.ts [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    },
+    {
+      value: goingToEventCount,
+      text: 'Cumul des participants',
```
Comment (CruuzAzul) [inline modules/meetup/queries/numbers.api.ts:73]:
```suggestion
      text: 'Participants au total',
```

**Review by CruuzAzul — APPROVED** [review]

**Comment by CruuzAzul** [comment]
I wonder if it shouldn't be displayed on the home page above the sponsors.... wyt ?

File: modules/about/Numbers.module.css [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
+  width: 180px;
+}
+
+.list li div:first-of-type {
+  font-size: 3rem;
+  color: var(--main-color);
+  font-weight: 500;
```
Comment (CruuzAzul) [inline modules/about/Numbers.module.css:19]: 700 or 900 maybe ?

```suggestion
  font-weight: 900;
```

**Comment by Slashgear** [comment]
@CruuzAzul Clearly that something we could add on homepage, maybe we could ask @jsulpis if he has idea to make it look nice. In about page, this design seems to fit but I am bit afraid it won't match homepage layout

**Comment by jsulpis** [comment]
> @CruuzAzul Clearly that something we could add on homepage, maybe we could ask @jsulpis if he has idea to make it look nice. In about page, this design seems to fit but I am bit afraid it won't match homepage layout

Hi, nice addition !
Indeed I think it's more common to see these figures in the homepage. I just copied/pasted the HTML of the preview (yes 😅) from the about page to the home page, put it in an `article` tag to center it, and it looks ok to me. I also changed the `line-height` of the figures to 1.2 to bring them closer to their caption, and used `color: var(--font-color-strong)` for better blending with the rest of the page.

btw it should be quite easy to animate the figures when they become visible, if you want. Motion one's [inView](https://motion.dev/dom/in-view) function should do the trick.

image

**Comment by Slashgear** [comment]
@jsulpis I am on it ? 
Did you have an animation in mind ? 
Opacity fade in maybe ?

**Comment by jsulpis** [comment]
> @jsulpis I am on it ? Did you have an animation in mind ? Opacity fade in maybe ?

I had in mind the classic counter, something like that: https://codepen.io/Amaik/pen/ExaaLgE
but faster (maybe ~2sec), and I think an `ease-out` timing function would be good

**Comment by Slashgear** [comment]
I will open dedicate PR with animation, I need to properly setup motion in the app :)

**Pull request merged**